### PR TITLE
Book verification

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: the_kybra_book/src/cross_canister.md [hunk 1]
```diff
@@ -105,88 +113,100 @@ Let's add to our example code and explore adding some practical result-based err
...
+                "_": lambda _: {
+                    "Err": f"Ok and Err both do not exist in {str(transfer_result)}"
+                },
+            },
+        )
+
+    return match(
+        call_result,
+        {
+            "Ok": lambda ok: handle_transfer_result_ok(ok),
```
Comment (lastmjs) [inline the_kybra_book/src/cross_canister.md:205]: Don't call the function here, please do a pass to try and find all of those

File: the_kybra_book/src/cross_canister.md [hunk 2]
```diff
@@ -105,88 +113,100 @@ Let's add to our example code and explore adding some practical result-based err
...
-    return {
-        'ok': transfer_result['ok']
-    }
+    def handle_transfer_result_ok(transfer_result: TransferResult) -> PayoutResult:
+        return match(
+            transfer_result,
+            {
+                "Ok": lambda ok: {"Ok": ok},
+                "Err": lambda err: {"Err": str(err)},
+                "_": lambda _: {
```
Comment (lastmjs) [inline the_kybra_book/src/cross_canister.md:196]: Git rid of the underscore

File: the_kybra_book/src/http.md [hunk 3]
```diff
@@ -93,24 +110,28 @@ def eth_get_balance(ethereum_address: str) -> Async[JSON]:
...
+            "method": {"post": None},
+            "headers": [],
+            "body": f'{{"jsonrpc":"2.0","method":"eth_getBalance","params":["{ethereum_address}","earliest"],"id":1}}'.encode(
+                "utf-8"
+            ),
+            "transform": {"function": (ic.id(), "eth_transform"), "context": bytes()},
+        }
+    ).with_cycles(cycle_cost_total)
+
+    def handle_http_result_err(err: str):
```
Comment (lastmjs) [inline the_kybra_book/src/http.md:126]: We don't need the function here, we can just do the ic.trap(err)

File: the_kybra_book/src/http.md [hunk 4]
```diff
@@ -122,30 +143,31 @@ def eth_get_block_by_number(number: nat32) -> Async[JSON]:
...
+        if err:
+            ic.trap(err)
+        ic.trap("http_result had an error")
```
Comment (lastmjs) [inline the_kybra_book/src/http.md:162]: We don't need the function here, we can just do the ic.trap(err)

File: the_kybra_book/src/reference/call_apis/notify_with_payment_128.md [hunk 5]
```diff
@@ -7,8 +7,14 @@ Examples:
 -   [cycles](https://github.com/demergent-labs/kybra/tree/main/examples/cycles)
 
 ```python
-from kybra import NotifyResult, Principal, update
-from src.cycles.types import Cycles
+from kybra import nat, NotifyResult, Principal, Service, service_update, update
+
+
+class Cycles(Service):
+    @service_update
```
Comment (lastmjs) [inline the_kybra_book/src/reference/call_apis/notify_with_payment_128.md:14]: Why are there two decorators?

File: the_kybra_book/src/reference/management_canister/http_request.md [hunk 6]
```diff
@@ -32,12 +36,12 @@ def xkcd() -> Async[HttpResponse]:
         }
     ).with_cycles(cycle_cost_total)
 
-    if http_result.err is not None:
-        if http_result.err:
-            ic.trap(http_result.err)
-        ic.trap("http_result had an error")
+    def handle_http_result_err(err: str):
```
Comment (lastmjs) [inline the_kybra_book/src/reference/management_canister/http_request.md:39]: This def is not needed

File: the_kybra_book/src/update_methods.md [hunk 7]
```diff
@@ -90,27 +98,32 @@ Here's an example of how to trap and ensure atomic changes to your database:
...
+
 @update
 def set_many(entries: list[Entry]) -> void:
     for entry in entries:
-        result = db.insert(entry['key'], entry['value'])
+        result = db.insert(entry["key"], entry["value"])
 
-        if result.err is not None:
-            ic.trap(str(result.err))
+        if result.Err is not None:
```
Comment (lastmjs) [inline the_kybra_book/src/update_methods.md:125]: Should use a match

File: the_kybra_book/src/update_methods.md [hunk 8]
```diff
@@ -120,27 +133,32 @@ There is a limit to how much computation can be done in a single call to an upda
...
 
+
 @update
 def set_many(num_entries: nat64) -> void:
     for i in range(num_entries):
         result = db.insert(str(i), str(i))
 
-        if result.err is not None:
-            ic.trap(str(result.err))
+        if result.Err is not None:
```
Comment (lastmjs) [inline the_kybra_book/src/update_methods.md:160]: Should use a match

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: the_kybra_book/src/http.md (hunk 3 above)
Comment (bdemann) [reply the_kybra_book/src/http.md:126]: I need to talk to Jordan about this one

Comment (bdemann) [reply the_kybra_book/src/http.md:126]: I talked to Jordan and he informed me that the old nested if was only to make Pylance happy, now that we have the match we don't need it and we can simplify to just trapping immediately with the error.

**Review by lastmjs — APPROVED** [review]

**Pull request merged**